src/SignalR/clients/csharp/Client/src/HttpConnectionFactory.cs

```diff
@@ -36,17 +45,46 @@ public HttpConnectionFactory(IOptions<HttpConnectionOptions> options, ILoggerFac
                 throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            _transferFormat = hubProtocol.TransferFormat;
             _httpConnectionOptions = options.Value;
             _loggerFactory = loggerFactory;
         }
 
-        /// <inheritdoc />
-        public async Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, CancellationToken cancellationToken = default)
+        /// <summary>
+        /// Creates a new connection to an <see cref="HttpEndPoint"/>.
+        /// </summary>
+        /// <param name="httpEndPoint">The <see cref="HttpEndPoint"/> to connect to.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        /// <returns>
+        /// A <see cref="ValueTask{TResult}"/> that represents the asynchronous connect.
+        /// The <see cref="Task{TResult}.Result"/> property returns a <see cref="ConnectionContext"/> for the new connection.
+        /// </returns>
+        public async ValueTask<ConnectionContext> ConnectAsync(EndPoint httpEndPoint, CancellationToken cancellationToken = default)
         {
-            var connection = new HttpConnection(_httpConnectionOptions, _loggerFactory);
+            if (httpEndPoint == null)
+            {
+                throw new ArgumentNullException(nameof(httpEndPoint));
+            }
+
+            var castedHttpEndPoint = httpEndPoint as HttpEndPoint;
+            if (castedHttpEndPoint == null)
+            {
+                throw new NotSupportedException($"The provided {nameof(EndPoint)} must be of type {nameof(HttpEndPoint)}.");
+            }
+
+            if (_httpConnectionOptions.Url != null && _httpConnectionOptions.Url != castedHttpEndPoint.Url)
+            {
+                throw new InvalidOperationException($"If {nameof(HttpConnectionOptions)}.{nameof(HttpConnectionOptions.Url)} was set, it must match the {nameof(HttpEndPoint)}.{nameof(HttpEndPoint.Url)} passed to {nameof(ConnectAsync)}.");
+            }
+
+            var shallowCopiedOptions = ShallowCopyHttpConnectionOptions(_httpConnectionOptions);
```

Comment:
Comments as to why we copy here.

Reply:
For my own knowledge, where do these options get modified? Feels like a bad pattern to be modifying options in general. Should just store internal state rather than needing to shallow copy.